sdk/Pulumi.Tests/Provider/PropertyValueTests.cs

```diff
@@ -299,6 +315,24 @@ public async Task DeserializingWrappedOutputSecretWorks()
         Assert.True(data.IsKnown);
     }
 
+    [Fact]
+    public async Task DeserializingWrappedUnknownOutputSecretWorks()
+    {
+        var serializer = CreateSerializer();
+        // secretOutput = Secret(Output(Unknown))
+        var secretOutput = new PropertyValue(
+            new PropertyValue(new OutputReference(
+                value:  PropertyValue.Computed,
+                dependencies: ImmutableArray<Urn>.Empty)));
```

Comment:
Shouldn't "secret" be on the OutputReference object?

Reply:
Ah no, this just needs to wrap in the double PropertyValue as it's doing.

Reply:
In deserialize there are both cases covered.

- Secret(Output(...)) being deserialized into Input<T>
- Output(Secret(...)) being deserialized into Input<T>

There is also already: 

- DeserializingWrappedSecretInOutputWorks which deserializes Output(Secret("Hello"))
- DeserializingWrappedOutputSecretWorks which deserializes Secret(Output("Hello"))

I just assumed that both value can be unknown

Reply:
Yeh either case could be unknown.